### PR TITLE
Remove FXIOS-5064/8318/3960 [v123] LegacyThemeManager removal

### PR DESCRIPTION
**Pull request by adudenamedruby**
## :scroll: Tickets
- Remove LegacyThemeManager [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5064) [Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/11820)

- Handle visual effects theming [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5064) [Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/12106)

- Fix tab tray header view bug [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5064) [Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18434)

## :bulb: Description
Bye Felicia!

Note, this PR requires that #18429 gets merged first.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: BrowserKit/Sources/Common/Theming/DefaultThemeManager.swift [hunk 1]
```diff
@@ -104,9 +110,9 @@ public final class DefaultThemeManager: ThemeManager, Notifiable {
     public func setSystemTheme(isOn: Bool) {
         userDefaults.set(isOn, forKey: ThemeKeys.systemThemeIsOn)
 
-        if isOn {
+        if systemThemeIsOn {
```
Comment (OrlaM) [inline BrowserKit/Sources/Common/Theming/DefaultThemeManager.swift:113]: It's better to keep isOn here, doing a file read here adds no value.

File: BrowserKit/Sources/Common/Theming/DefaultThemeManager.swift [hunk 2]
```diff
@@ -130,6 +135,14 @@ public final class DefaultThemeManager: ThemeManager, Notifiable {
         brightnessChanged()
     }
 
+    public func brightnessChanged() {
+        if automaticBrightnessIsOn {
+            updateThemeBasedOnBrightness()
+        } else {
+            systemThemeChanged()
+        }
```
Comment (OrlaM) [inline BrowserKit/Sources/Common/Theming/DefaultThemeManager.swift:143]: I know this was part of the old code but you can actually remove this. If the brightness changes but the user doesn't have the automatic brightness change enabled the correct response is to do nothing.  No reason to set the theme at this point, it should be redundant.

**Review by OrlaM — APPROVED** [review]
A couple of nits, but otherwise LGTM 👍

**Comment by adudenamedruby** [comment]
@Mergifyio backport release/v123

**Comment by adudenamedruby** [comment]
@Mergifyio backport release/v122

**Pull request merged**